### DISCUSSION TITLE
Welcome to esphome-jk-bms Discussions!

### DISCUSSION THREAD

**Discussion by txubelaxu** [txubelaxu/esphome-jk-bms#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by testpaul999** [comment]
Hi,

have test the new esp32-ble-battery_number_001.yaml.
It runs good all parameters are there.
I have modify a little bit for more structure. 
[jk-bms-test.yaml.txt](https://github.com/txubelaxu/esphome-jk-bms/files/14670653/jk-bms-test.yaml.txt)
all switches are together and so...

regards

**Comment by jogybaer0815** [comment]
i run into an problem with 1 new JK PB2A16S15P
i get 2 new bms and set new Passwords. One is ok but the other locked me out. Password is not recognized and it's so tedious to deal with the Ali shop. I'm now not able to change settings.
I have seen that the "Monitor a JK-BMS via UART-TTL" version can pull the Password but the bt version does not.
Is there a way to read the password via BT?
Thanks

**Reply by txubelaxu** [reply]
Hi @jogybaer0815,

I think that BT is not showing info from frame type number 3.

Have you tested RS485?

Regards

**Reply by concierge-at** [reply]
Okay thank you, an wich code is recommend for the new jk bms for rs 485? 
Is there an circuit for rs485 connection? I try 485 on the old bms without succsess😭🫣

**Comment by jogybaer0815** [comment]
RS485 is showing only "1234" but this is not the real password. That is only linking pass for bt connection.
Can you investigate?
Thanks in advance

**Comment by jogybaer0815** [comment]
> 
> I think that BT is not showing info from frame type number 3.
>...
but the esp32-example only has text_sensor password  - the esp32-ble-versions does not include this text_sensor

**Comment by concierge-at** [comment]
Hello everyone, thanks for providing the code.
I got it running with BMS HW version 15. The BMS ESP connection also works. The ESP HA connection also works. 
But unfortunately after a few minutes the ESP no longer receives any data. I only use a power supply and the temperature sensors for a test. Only a restart of the ESP provides fresh data that the HA automation is running again. Is there already a solution to the connection loss? that it remains stable. Thanks in advance
![JK-Loss](https://github.com/txubelaxu/esphome-jk-bms/assets/135752342/064836af-38c8-4ad7-a576-212160aa9074)
[jk-bms-ble-new.yaml.txt](https://github.com/user-attachments/files/15662285/jk-bms-ble-new.yaml.txt)

**Comment by jogybaer0815** [comment]
Bluetooth is not stable for me. I have added a restart button to restart ESP. Sometimes it will help to disable and reable BT.
Much better and stable is the RS485 connection.

**Comment by concierge-at** [comment]
Hello thank you,
Wich code is recomendet for the new jk wiith rs485 i will try it. The old one wars impossible to connect on rs 485 for me. 
I am not acprogrammer i am on the beginning of esp scratching lightly on the surface, there are avlotbof yamel files. Please help me to choose the right one
Thank you

**Comment by jogybaer0815** [comment]
https://github.com/txubelaxu/esphome-jk-bms/blob/main/esp32-example-jkpb-rs485_1master_1slave.yaml
use rs485 file (for 1 or 6 slaves) and adress BMS jumper from 1 to x to act as slaves.

edit: link corrected

**Reply by concierge-at** [reply]
> https://github.com/txubelaxu/esphome-jk-bms/blob/main/esp32-example-jkpb-rs485_1master_1slave.yaml Verwenden Sie eine RS485-Datei (für 1 oder 6 Slaves) und adressieren Sie den BMS-Jumper von 1 bis x, damit sie als Slaves fungieren.
> 
> edit: Link korrigiert

Thank you verry much, Sorry for my english its not my native language.
i have took the [esp32-beispiel-jkpb-rs485_1master_1slave.yaml](https://github.com/txubelaxu/esphome-jk-bms/blob/main/esp32-example-jkpb-rs485_1master_1slave.yaml) and delete the second device als Cell 17-24

i allso have found the Pins for the communication in drawing, thank you. i will test them its my spare when the old one will not work again thank you

**Comment by jogybaer0815** [comment]
Du kannst aber auch auf deutsch schreiben und darunter die Google Übersetzung 
-
But you can also write in German and use Google translation underneath

**Comment by vuong244** [comment]
Tôi mới nhập môn cái này, và tôi không biết bắt đầu từ đâu, tôi đã kết nối các mạch lại với nhau, nhưng tôi không biết chạy code vào mạch esp32 như thế nào
